extensions-core/namespace-lookup/src/main/java/io/druid/server/namespace/cache/NamespaceExtractionCacheManager.java

```diff
@@ -225,38 +225,56 @@ public String apply(
       log.debug("Deleting %d namespaces: %s", differentNamespaces.size(), differentNamespaces);
     }
     for (String namespaceName : differentNamespaces) {
-      final NamespaceImplData implDatum = implData.get(namespaceName);
-      if (implDatum == null) {
-        // Delete but we don't have it?
-        log.wtf("Asked to delete something I just lost [%s]", namespaceName);
-        continue;
-      }
-      delete(namespaceName);
+      deleteIndividual(namespaceName);
     }
     for (final ExtractionNamespace namespace : namespaces) {
-      final NamespaceImplData implDatum = implData.get(namespace.getNamespace());
-      if (implDatum == null) {
-        // New, probably
-        schedule(namespace);
-        continue;
-      }
-      if (!implDatum.enabled.get()) {
-        // Race condition. Someone else disabled it first, go ahead and reschedule
-        schedule(namespace);
-        continue;
-      }
+      scheduleOrUpdateIndividual(namespace);
+    }
+  }
 
-      // Live one. Check if it needs updated
-      if (implDatum.namespace.equals(namespace)) {
-        // skip if no update
-        continue;
-      }
-      if (log.isDebugEnabled()) {
-        log.debug("Namespace [%s] needs updated to [%s]", implDatum.namespace, namespace);
-      }
-      removeNamespaceLocalMetadata(implDatum);
+  // return value means actually delete or not
+  public boolean deleteIndividual(
+      String namespaceName
+  )
+  {
+    final NamespaceImplData implDatum = implData.get(namespaceName);
+    if (implDatum == null) {
+      // Delete but we don't have it?
+      log.wtf("Asked to delete something I just lost [%s]", namespaceName);
+      return false;
+    }
+    delete(namespaceName);
+    return true;
+  }
+
+  // return value means actually schedule or not
+  public boolean scheduleOrUpdateIndividual(
```

Comment:
you should be able to leave this as `scheduleOrUpdate` and just have the method parameter be a single `ExtractionNamespace` instead of a `Collection`.